ideascube/management/commands/tags.py

```diff
@@ -121,3 +136,44 @@ def list(self, options):
         print(row.format('.' * 40, '.' * 40, '.' * 40))
         for tag in Tag.objects.order_by('slug'):
             print(row.format(tag.name, tag.slug, self._count(tag.name)))
+
+    def sanitize(self, options):
+        for relation in TaggedItem.objects.all():
+            tag = relation.tag
+            new_tag_name = sanitize_tag_name(tag.name)
+            if not new_tag_name:
+                # The tag was just a punctuation or a space.
+                relation.delete()
+                continue
+            # Get tags with the same case-insensitive final name.
+            tags = Tag.objects.filter(name__iexact=new_tag_name).order_by('id')
+
+            if not tags.count():
+                # There is none.
+                # This can happen if we change the tag name.
+                # Then, we just want to rename the tag.
+                tag.name = new_tag_name
+                tag.save()
+            else:
+                new_tag = tags[0]
+                if tag == new_tag:
+                    continue
```

Comment:
Right, I understand what those lines do. What I don't get is why they are here in this form.

They look awfully specific, but I just can't make sense out of them.

For example, what about the other items of the `tags` list? Why ignore them?